exporter/signalfxexporter/exporter.go

```diff
@@ -146,6 +145,33 @@ func newSignalFxExporter(
 	}, nil
 }
 
+func getHTTPClient(config *Config) (*http.Client, error) {
+	defaultMaxConns := defaultMaxIdleConns
+	if config.MaxIdleConns == nil {
+		if config.MaxConnections != nil {
+			config.MaxIdleConns = config.MaxConnections
+		} else {
+			config.MaxIdleConns = &defaultMaxConns
+		}
+	}
+	if config.MaxIdleConnsPerHost == nil {
+		if config.MaxConnections != nil {
+			config.MaxIdleConnsPerHost = config.MaxConnections
+		} else {
+			config.MaxIdleConnsPerHost = &defaultMaxConns
+		}
+	}
+
+	// Using another variable to avoid the local header changes from reflecting
+	// into actual config.
+	httpClientSettings := config.HTTPClientSettings
+	// To avoid overriding client headers to request header storing access_token
+	// when access_token_passthrough is true.
+	httpClientSettings.Headers = map[string]string{}
```

Comment:
Aren't we completely discard `Headers` config here? I believe `pushMetricsDataForToken` takes care of setting token from access_token_passthrough

Reply:
We are not completely discarding the headers here. We have assigned `config.HTTPClientSettings` to different variable (`httpClientSettings`) and we are discarding headers of this variable. So the headers in the config will be there and are not discarded.
  You are right the `pushMetricsDataForToken` function is assigning [request headers](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/bb538051fb03070d1683223dc10c381173fb9328/exporter/signalfxexporter/dpclient.go#L131) as per the access_token_passthrough, but after that when we are calling [client.Do(req)](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/bb538051fb03070d1683223dc10c381173fb9328/exporter/signalfxexporter/dpclient.go#L140), it is calling the [roundTrip](https://github.com/open-telemetry/opentelemetry-collector/blob/4d0b6e4f891efca70a85a28ca6db74458b0dc935/config/confighttp/confighttp.go#L192) function of HTTPClient and in that the request headers are overridden by the client headers. So to avoid this we have assigned empty headers to client here.

Reply:
Got it. Thanks for explanation